core/src/main/java/org/opensearch/sql/calcite/udf/UserDefinedAggFunction.java

```diff
@@ -5,11 +5,34 @@
 
 package org.opensearch.sql.calcite.udf;
 
-public interface UserDefinedAggFunction<S extends Accumulator> {
+public interface UserDefinedAggFunction<S extends UserDefinedAggFunction.Accumulator> {
+  /**
+   * @return {@link Accumulator}
+   */
   S init();
 
+  /**
+   *
+   * @param {@link Accumulator}
+   * @return final result
+   */
   Object result(S accumulator);
 
-  // Add values to the accumulator
+  /**
+   * Add values to the accumulator. Notice some init argument will also be here like the 50 in Percentile(field, 50).
+   * @param acc {@link Accumulator}
+   * @param values the value to add to accumulator
+   * @return {@link Accumulator}
+   */
   S add(S acc, Object... values);
+
+  /**
+   * Maintain the state when {@link UserDefinedAggFunction} add values
+   */
+  interface Accumulator {
+    /**
+      * @return the final aggregation value
```

Comment:
Have you run `./gradlew spotlessApply` before pushing?

Reply:
Forget to run. Just rerun the push.